src/state_machine.zig

```diff
@@ -516,18 +517,102 @@ fn equal_48_bytes(a: [48]u8, b: [48]u8) bool {
 
 const testing = std.testing;
 
+test "create/lookup accounts" {
+    var arena = std.heap.ArenaAllocator.init(std.heap.page_allocator);
+    defer arena.deinit();
+    const allocator = &arena.allocator;
+
+    var accounts = [_]Account{
+        std.mem.zeroInit(Account, .{ .id = 1, .flags = .{ .padding = 1 } }), //.reserved_flag_padding
```

Comment:
As we chatted about, we could encode the expected error enum literal into a tuple struct (e.g. `enum, test value`). This way the compiler can help us out if enums get renamed, it's more visible than a comment, and we can then make our test code below more generic with a loop. This also makes it easy to add more test cases.